packages/client/src/vscode/services.ts

```diff
@@ -100,18 +100,24 @@ export const initServices = async (vscodeApiConfig: VscodeApiConfig, instruction
     if (!(envEnhanced.vscodeInitialising ?? false)) {
         if (envEnhanced.vscodeApiInitialised ?? false) {
             instructions.logger?.debug('Initialization of vscode services can only performed once!');
+
+            if (vscodeApiConfig.userConfiguration?.json !== undefined) {
+                await updateUserConfiguration(vscodeApiConfig.userConfiguration?.json);
+            }
         } else {
             envEnhanced.vscodeInitialising = true;
             instructions.logger?.debug(`Initializing vscode services. Caller: ${instructions.caller ?? 'unknown'}`);
 
+            if (vscodeApiConfig.userConfiguration?.json !== undefined) {
+                await initUserConfiguration(vscodeApiConfig.userConfiguration?.json);
+            }
```

Comment:
I guess we may want a way to initialize the default configuration without overriding it each time :thinking:

Reply:
Ah, yes, sorry, I make this mistake over and over. We have to merge what is already there.

Reply:
`getUserConfiguration` does not return (await hangs forever) before any service init is done. Is this expected? That call above is only done once, btw.

Reply:
Yes, it is expected

I'm not sure of the point of even merging the configuration :thinking:

Reply:
> I'm not sure of the point of even merging the configuration 🤔

Wasn't that the point of your first comment? "I guess we may want a way to initialize the default configuration without overriding it each time"  I am confused.

Reply:
I mean just calling initUserConfiguration when the service aren't initialized

I don't think I would want the configuration to be "restored" if I call initServices a second time :thinking:

Reply:
Ok, then the last commit matches what you expect.

Reply:
`vscodeApiInitialised` is stored in the "enhanced" global monaco environment and once services are initialized that code block is not hit again.

Reply:
I'm still concerned by the fast that you're updating the configuration when `vscodeApiInitialised` is true. I'm not sure when it's supposed to be called, but:
- I would change the file formatting
- It would change the property order
- It would reset user settings if the user changes it

I'm not sure about the point of it :thinking: if the developper want to keep the control of a settings, they can just use the defaultConfiguration in the workbench construct options. Initializing the user settings only exists so that the file is not empty, and to showcase what property the user may want to change

Reply:
Ok, then let's only do user config init before services are configured and only if set in the config. This is only done once. Even if you recall the `initServices`.
What the previous code did was wrong, because it always called `updateUserConfiguration`. I really should add a unit test for this actually.

Reply:
Ok, changes and unit test are up ⬇️